File: pkg/dev_compiler/test/codegen_expected/corelib/map_values_test.js
dart_library.library('corelib/map_values_test', null, /* Imports */[
  'dart_sdk',
  'expect'
], function load__map_values_test(exports, dart_sdk, expect) {
  'use strict';
  const core = dart_sdk.core;
  const dart = dart_sdk.dart;
  const dartx = dart_sdk.dartx;
  const expect$ = expect.expect;
  const map_values_test = Object.create(null);
  let MapOfString$int = () => (MapOfString$int = dart.constFn(core.Map$(core.String, core.int)))();
  let MapOfString$bool = () => (MapOfString$bool = dart.constFn(core.Map$(core.String, core.bool)))();
  let VoidTodynamic = () => (VoidTodynamic = dart.constFn(dart.definiteFunctionType(dart.dynamic, [])))();
  let const$;
  let const$0;
  map_values_test.main = function() {
    let map1 = dart.map({foo: 42, bar: 499}, core.String, core.int);
    let map2 = dart.map();
    let map3 = const$ || (const$ = dart.const(dart.map({foo: 42, bar: 499}, core.String, core.int)));
    let map4 = const$0 || (const$0 = dart.const(dart.map()));
    let map5 = MapOfString$int().new();
    map5[dartx.set]("foo", 43);
    map5[dartx.set]("bar", 500);
    let map6 = MapOfString$bool().new();
    expect$.Expect.isTrue(core.Iterable.is(map1[dartx.values]));
    expect$.Expect.isFalse(core.List.is(map1[dartx.values]));
    expect$.Expect.equals(2, map1[dartx.values][dartx.length]);
    expect$.Expect.equals(42, map1[dartx.values][dartx.first]);
    expect$.Expect.equals(499, map1[dartx.values][dartx.last]);
    expect$.Expect.isTrue(core.Iterable.is(map2[dartx.values]));
    expect$.Expect.isFalse(core.List.is(map2[dartx.values]));
    expect$.Expect.equals(0, map2[dartx.values][dartx.length]);
    expect$.Expect.isTrue(core.Iterable.is(map3[dartx.values]));
    expect$.Expect.isFalse(core.List.is(map3[dartx.values]));
    expect$.Expect.equals(2, map3[dartx.values][dartx.length]);
    expect$.Expect.equals(42, map3[dartx.values][dartx.first]);
    expect$.Expect.equals(499, map3[dartx.values][dartx.last]);
    expect$.Expect.isTrue(core.Iterable.is(map4[dartx.values]));
    expect$.Expect.isFalse(core.List.is(map4[dartx.values]));
    expect$.Expect.equals(0, map4[dartx.values][dartx.length]);
    expect$.Expect.isTrue(core.Iterable.is(map5[dartx.values]));
    expect$.Expect.isFalse(core.List.is(map5[dartx.values]));
    expect$.Expect.equals(2, map5[dartx.values][dartx.length]);
    expect$.Expect.isTrue(map5[dartx.values][dartx.first] == 43 || map5[dartx.values][dartx.first] == 500);
    expect$.Expect.isTrue(map5[dartx.values][dartx.last] == 43 || map5[dartx.values][dartx.last] == 500);
    expect$.Expect.notEquals(map5[dartx.values][dartx.first], map5[dartx.values][dartx.last]);
    expect$.Expect.isTrue(core.Iterable.is(map6[dartx.values]));
    expect$.Expect.isFalse(core.List.is(map6[dartx.values]));
    expect$.Expect.equals(0, map6[dartx.values][dartx.length]);
  };
  dart.fn(map_values_test.main, VoidTodynamic());
  // Exports:
  exports.map_values_test = map_values_test;
});

File: tests/corelib/map_values_test.dart
// Copyright (c) 2011, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

import "package:expect/expect.dart";

main() {
  var map1 = {"foo": 42, "bar": 499};
  var map2 = {};
  var map3 = const {"foo": 42, "bar": 499};
  var map4 = const {};
  var map5 = new Map<String, int>();
  map5["foo"] = 43;
  map5["bar"] = 500;
  var map6 = new Map<String, bool>();

  Expect.isTrue(map1.values is Iterable);
  Expect.isFalse(map1.values is List);
  Expect.equals(2, map1.values.length);
  Expect.equals(42, map1.values.first);
  Expect.equals(499, map1.values.last);

  Expect.isTrue(map2.values is Iterable);
  Expect.isFalse(map2.values is List);
  Expect.equals(0, map2.values.length);

  Expect.isTrue(map3.values is Iterable);
  Expect.isFalse(map3.values is List);
  Expect.equals(2, map3.values.length);
  Expect.equals(42, map3.values.first);
  Expect.equals(499, map3.values.last);

  Expect.isTrue(map4.values is Iterable);
  Expect.isFalse(map4.values is List);
  Expect.equals(0, map4.values.length);

  Expect.isTrue(map5.values is Iterable);
  Expect.isFalse(map5.values is List);
  Expect.equals(2, map5.values.length);
  Expect.isTrue(map5.values.first == 43 || map5.values.first == 500);
  Expect.isTrue(map5.values.last == 43 || map5.values.last == 500);
  Expect.notEquals(map5.values.first, map5.values.last);

  Expect.isTrue(map6.values is Iterable);
  Expect.isFalse(map6.values is List);
  Expect.equals(0, map6.values.length);
}
